src/os/filestore/FileStore.cc

```diff
@@ -3423,6 +3423,7 @@ int FileStore::read(
   if (got < 0) {
     dout(10) << __FUNC__ << ": (" << cid << "/" << oid << ") pread error: " << cpp_strerror(got) << dendl;
     lfn_close(fd);
+    ceph_assert(!m_filestore_fail_eio || l != -EIO);
```

Comment:
@gregsfortytwo This was removed so that deep-scrub can report the error.  It can reported to the user and eventually repaired.  Otherwise, you have to delete the object or switch the config to top the crashing and repair it.

Reply:
@dzafman we’ve got a problem then because this assert is the very most common one — a normal read from the primary!
So in practice we don’t ever assert on EIO, the config option doesn’t work as expected, and our claiming it works is a lie that leads to user bug reports.

Any suggestions for how to best resolve that inconsistency?